backend/api/publishing.py

```
@@ -394,6 +894,167 @@ async def save_scenario_draft(
+        
+    except HTTPException:
+        raise
+    except Exception as e:
+        db.rollback()
+        debug_log(f"Failed to update scenario status: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Failed to update scenario status: {str(e)}")
+
+@router.delete("/unique/{unique_id}")
+async def delete_scenario_by_unique_id(
+    unique_id: str,
+    db: Session = Depends(get_db),
```

Ruff (0.14.0)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)